File: src/netconsole/models/agent.py
from __future__ import annotations

from dataclasses import dataclass, field
from enum import StrEnum
from typing import Any


class AgentStatus(StrEnum):
    UNKNOWN = "UNKNOWN"
    ONLINE = "ONLINE"
    OFFLINE = "OFFLINE"
    UNAUTHORIZED = "UNAUTHORIZED"
    DISABLED = "DISABLED"


class AgentAuthenticationType(StrEnum):
    NONE = "none"
    TOKEN = "token"


@dataclass(frozen=True)
class AgentConfig:
    agent_id: str
    name: str
    base_url: str
    enabled: bool = True
    authentication_type: AgentAuthenticationType = AgentAuthenticationType.NONE
    credential_reference: str = ""
    tags: list[str] = field(default_factory=list)
    note: str = ""
    created_at: str = ""
    updated_at: str = ""
    archived_at: str = ""


@dataclass(frozen=True)
class AgentRuntimeSnapshot:
    agent_id: str
    status: AgentStatus = AgentStatus.UNKNOWN
    last_seen_at: str = ""
    last_checked_at: str = ""
    latency_ms: int | None = None
    version: str = ""
    platform: str = ""
    architecture: str = ""
    capabilities: dict[str, Any] = field(default_factory=dict)
    last_error_code: str = ""
    last_error_message: str = ""
    updated_at: str = ""

File: frontend/src/types/agent.ts
export type AgentStatus = 'UNKNOWN' | 'ONLINE' | 'OFFLINE' | 'UNAUTHORIZED' | 'DISABLED'
export type AgentAuthenticationType = 'none' | 'token'

export interface AgentItem {
  agent_id: string
  name: string
  base_url: string
  enabled: boolean
  authentication_type: AgentAuthenticationType
  has_credential: boolean
  tags: string[]
  note: string
  created_at: string
  updated_at: string
  status: AgentStatus
  last_seen_at: string
  last_checked_at: string
  latency_ms: number | null
  version: string
  platform: string
  architecture: string
  capabilities: Record<string, unknown>
  last_error_code: string
  last_error_message: string
}

export interface AgentFormValue {
  name: string
  base_url: string
  enabled: boolean
  authentication_type: AgentAuthenticationType
  token?: string
  tags: string[]
  note: string
}

export interface AgentProbeResult {
  remote_agent_id: string
  remote_name: string
  version: string
  platform: string
  architecture: string
  capabilities: Record<string, unknown>
  latency_ms: number
}

export interface AgentSocketEvent {
  type: string
  agent_id?: string
  agents?: AgentItem[]
}
